definitions/scenarios/self_upgrade.rb

```diff
@@ -47,12 +57,12 @@ def req_repos_to_update_pkgs
       end
     end
 
-    def upstream_target_version
-      if feature(:katello_install)
-        return foreman_version_by_katello(target_version)
-      else
-        target_version
-      end
+    def upgrade_allowed?
+      Gem::Version.new(current_version) == Gem::Version.new(target_version)
+    end
```

Comment:
don't we need `current_version == target_version - 1` to allow  upgrades?

Reply:
oooh, wait. you want to say "if you have the foreman maintain that supports the upgrade to `current_version`, your foreman maintain is too old to upgrade to the next version and thus this is the only time you're allowed to install "next" foreman maintain"?

Reply:
Yessir.

Reply:
I think I would have been less confused if the method would be `self_upgrade_required?`.

Reply:
Is it required though? Or simply possible? I think that naming ends up confusing me :)

Reply:
Well, I read is as "required" as otherwise it would have the wrong "target_release" and thus handle things wrongly?

Reply:
What other code does with this information is different than what is this method telling us. At least, that was my design desire - to have a method that could tell me if self-upgrade is possible.